### PR TITLE
Deprecate the `UriNormalize` filter

### PR DESCRIPTION
**Pull request by gsteel**
See #110

### CODE REVIEW

File: docs/book/v2/standard-filters.md [hunk 1]
```diff
@@ -1912,6 +1912,9 @@ as the result.
 
 ## UriNormalize
 
+CAUTION: **Deprecated**
+This filter is deprecated since version 2.36.0 and will be remove in version 3.0. There will not be a replacement.
```
Comment (froschdesign) [inline docs/book/v2/standard-filters.md]: > There will not be a replacement.

I think there should be a new filter for this?

**Review by Ocramius — APPROVED** [review]

File: docs/book/v2/standard-filters.md (hunk 1 above)
Comment (gsteel) [reply docs/book/v2/standard-filters.md]: Does #133 go far enough here? If so, I'll amend to reference this one after merging

**Comment by froschdesign** [comment]
@gsteel 
A deprecation requires the creation of or an entry in a migration guide. We have a good example in laminas-view: https://docs.laminas.dev/laminas-view/v2/migration/preparing-for-v3/ 😃

File: src/UriNormalize.php [hunk 2]
```diff
@@ -16,6 +16,8 @@
 use function str_contains;
 
 /**
+ * @deprecated This filter will be removed in version 3.0 without replacement
```
Comment (froschdesign) [inline src/UriNormalize.php]: This text should be the same as the deprecation message in the documentation. (see above)

File: docs/book/v2/migration/preparing-for-v3.md [hunk 3]
```diff
@@ -0,0 +1,55 @@
+# Preparing for Version 3
```
Comment (froschdesign) [inline docs/book/v2/migration/preparing-for-v3.md:1]: ❤️

**Review by froschdesign — CHANGES_REQUESTED** [review]
Great work on the migration guide! 👍🏻 

Only a minor change is required.

File: src/UriNormalize.php (hunk 2 above)
Comment (gsteel) [reply src/UriNormalize.php]: Thanks for the review @froschdesign

**Review by froschdesign — APPROVED** [review]

**Pull request merged**